lib/src/query_parameters/index_search_query.dart

```diff
@@ -0,0 +1,82 @@
+import '../filter_builder/_exports.dart';
+import '../matching_strategy_enum.dart';
+import 'search_query.dart';
+
+class IndexSearchQuery extends SearchQuery {
+  final String indexUid;
+
+  const IndexSearchQuery(
+    super.query, {
+    required this.indexUid,
+    super.offset,
+    super.limit,
+    super.page,
+    super.hitsPerPage,
+    super.filter,
+    super.filterExpression,
+    super.sort,
+    super.facets,
+    super.attributesToRetrieve,
+    super.attributesToCrop,
+    super.cropLength,
+    super.attributesToHighlight,
+    super.showMatchesPosition,
+    super.cropMarker,
+    super.highlightPreTag,
+    super.highlightPostTag,
+    super.matchingStrategy,
+  });
+
+  @override
+  Map<String, Object> toMap() {
+    return {
+      'indexUid': indexUid,
+      ...super.toMap(),
+    };
+  }
+
+  @override
+  IndexSearchQuery copyWith({
```

Comment:
Where this method `copyWith` is used?

Reply:
it will be used in the flutter ui to manipulate offsets/pagination automatically, with the user only passing other parameters when needed